vnext/Microsoft.ReactNative.SharedManaged/ReactPackageBuilderExtensions.cs

```diff
@@ -23,47 +35,57 @@ public void AddToPackageBuilder(IReactPackageBuilder packageBuilder)
       }
     }
 
-    private static readonly Lazy<Dictionary<string, AttributedModule>> m_attributedModules =
-      new Lazy<Dictionary<string, AttributedModule>>(InitAttributedModules, LazyThreadSafetyMode.PublicationOnly);
-
-    private static Dictionary<string, AttributedModule> InitAttributedModules()
+    private static Dictionary<string, AttributedModule> InitAttributedModules(Assembly assembly)
     {
       var initializers =
-        from typeInfo in typeof(ReactPackageBuilderExtensions).GetTypeInfo().Assembly.DefinedTypes
+        from typeInfo in assembly.DefinedTypes
         let moduleAttribute = typeInfo.GetCustomAttribute<ReactModuleAttribute>()
         where moduleAttribute != null
         select new AttributedModule { ModuleType = typeInfo.AsType(), ModuleAttribute = moduleAttribute };
       return initializers.ToDictionary(m => ReactModuleInfo.GetModuleName(m.ModuleType, m.ModuleAttribute));
     }
-
-    public static void AddAttributedModules(this IReactPackageBuilder packageBuilder)
+    
+    public static void AddAttributedModules(this IReactPackageBuilder packageBuilder, Assembly assembly)
     {
-      foreach (var attributedModule in m_attributedModules.Value)
+      var attributedModules = s_registeredAssemblies.GetOrAdd(assembly, asm => new Lazy<Dictionary<string, AttributedModule>>(() => InitAttributedModules(asm)));
+      foreach (var attributedModule in attributedModules.Value.Values)
       {
-        attributedModule.Value.AddToPackageBuilder(packageBuilder);
+        attributedModule.AddToPackageBuilder(packageBuilder);
       }
     }
 
+    public static void AddAttributedModules(this IReactPackageBuilder packageBuilder)
+    {
+      AddAttributedModules(packageBuilder, s_currentAssembly);
+    }
+
     public static bool TryAddAttributedModule(this IReactPackageBuilder packageBuilder, string moduleName)
```

Comment:
This function supposed to register only a single module out of all registered modules. I do not see how it may work because it only considers s_currentAssembly instead of all registered assemblies.

Reply:
All the new call sites use the overload that passes an assembly.
We still have projects out there in the wild that use this code as a "Shared Project" and have C# that use the extension method and assume 
```cs
// Includes any modules in this project
``


---
In reply to: [420396406](https://github.com/microsoft/react-native-windows/pull/4793#discussion_r420396406) [](ancestors = 420396406)

Reply:
I'll comment this class better to make the distinction.

---
In reply to: [420417179](https://github.com/microsoft/react-native-windows/pull/4793#discussion_r420417179) [](ancestors = 420417179,420396406)

Reply:
Could you mark it as Obsolete and add comment to use another method?

Reply:
I could but then all existing projects would break since they will get a warning.